lib/src/embedded/compiler/packet-transformer.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {Observable, Subject} from 'rxjs';
+import {filter} from 'rxjs/operators';
+
+/**
+ * Decodes arbitrarily-chunked buffers, for example
+ *   [ 0 1 2 3 4 5 6 7 ... ],
+ * into packets of set length in the form
+ *   +---------+------------- ...
+ *   | 0 1 2 3 | 4 5 6 7 ...
+ *   +---------+------------- ...
+ *   | HEADER  | PAYLOAD
+ *   +---------+------------- ...
+ * and emits the payload of each packet.
+ *
+ * Encodes payloads by attaching a header that describes the payload's length.
+ */
+export class PacketTransformer {
+  private readonly _read$ = new Subject<Buffer>();
```

Comment:
TODO: The underscore violates the style guide, but TBH this variable only exists to hide the Subject from public consumption. Need to find a name that captures this intent

Reply:
TO DONE